### DISCUSSION TITLE
Can I filter for conditional node styles?

### DISCUSSION THREAD

**Discussion by autoca** [freeplane/freeplane#734] [Help] [answered]
Can I filter the nodes in the map for conditional node styles?

So far, I could not find an option.
There is the option to make a custom filter with scripting but that is still beyond my capabilities.

Any help is apprechiated. MTIA and regards, a.

**Comment by EdoFro** [comment, marked as answer]
Which Freeplane version are you using?

I use version 1.10.5 and I have no problem filtering conditional styles. I do the same way I filter directly assigned styles.

![image](https://user-images.githubusercontent.com/35700575/195083361-1c9117a9-cb12-46de-b816-c1809d594319.png)

in this case "GroovyNode" is a conditional style.

It works using both alternatives: "Is equal to" and "contains"

and it shows only the nodes that have that style (conditional or directly assigned)

![image](https://user-images.githubusercontent.com/35700575/195084006-dde5fcf5-e3a4-4f77-9d21-1530b4e895d0.png)

**Reply by autoca** [reply]
Hi EdoFro,

thx for checking it out.
It seems that I only tested a more complicated case.

If the node already has a "normal" main style and then has conditional styles on top of it, then it does not include the nodes with these conditional styles in the filter anymore.

**Reply by EdoFro** [reply]
did you try it using the option "contains"?

I test it again and assigned the style "nextTask" to a node with conditional style "GroovyNode"

The filter didn't show it with the option "Is equal to", but it does show it with the option "Contains"

![image](https://user-images.githubusercontent.com/35700575/195117221-1264eace-019a-4a74-88e3-345d975d8a4c.png)

**Reply by autoca** [reply]
Hey EdoFro, thanks for not giving up,

Yes, that works. Sorry that I did not test it thoroughly enough; did not seem intuitive to me.
But glad it works anyway. Thank you again.

Regards, a.

P.S.:
How come that the .groovy style name is written in the details of all your nodes?
I assume that you did not do that by hand? But how did you do it?
